### ISSUE TITLE
[Question] Is that possible to run dom-to-image on NodeJS?

### ISSUE THREAD

**Issue by BrunoQuaresma** [1904labs/dom-to-image-more#48]
Hi folks, 

Just a quick question. Is that possible to run dom-to-image on NodeJS?

**Comment by SNDST00M** [comment]
You need `window.getComputedStyle`, so perhaps a headless browser instance would make it work

**Comment by BrunoQuaresma** [comment]
Yes, I'm already using a headless browser, but it is pretty slow, so I was trying to find a better alternative. Thank you.

**Comment by SNDST00M** [comment]
Crazy idea - just use https://github.com/jsdom/jsdom

**Comment by BrunoQuaresma** [comment]
I tried, but it didn't work. It gets an error related to canvas even after use node-canvas. 😞

**Comment by SNDST00M** [comment]
<p align="center"><img src="https://raw.githubusercontent.com/twitter/twemoji/master/assets/svg/1f1eb.svg" align="center" width="225px" /></p>

**Comment by meche-gh** [comment]
There was a perf-hitting extreme CSS optimisation added in #37 for SVGs.
- It got switched off for raster images in December - [`v2.9.5`](https://github.com/1904labs/dom-to-image-more/releases/tag/v2.9.5).
- A 5-times-faster (but less than half as good) CSS reducer was added for raster images about 5 months ago in #71 - [`v2.10.1`](https://github.com/1904labs/dom-to-image-more/releases/tag/v2.10.1)
- [`v2.13.1`](https://github.com/1904labs/dom-to-image-more/releases/tag/v2.13.1) - released yesterday - removed the stronger SVG optimization.

There's plans to bring it back in #92 as a much quicker post-processing op.
Choice of browser is also probably important. But I have no data to suggest anything more than "maybe MS Edge".

**Comment by yelban** [comment]
Curious to know if it works with node + jsdom

**Comment by anshumanv** [comment]
Revisiting here in 2024...

**Comment by IDisposable** [comment]
Not supported directly — the library needs a real DOM, layout, and a `<canvas>`/_SVG_ rasterizer, which Node doesn't provide. To render server-side, run it in a headless browser (Puppeteer/Playwright), or use `toSvg()` in the browser and rasterize the SVG on the server. Closing as out of scope; the headless-browser route is the supported path.

**Comment by IDisposable** [comment]
Not supported directly — the library needs a real DOM, layout, and a `<canvas>`/_SVG_ rasterizer, which Node doesn't provide. To render server-side, run it in a headless browser (Puppeteer/Playwright), or use `toSvg()` in the browser and rasterize the SVG on the server. Closing as out of scope; the headless-browser route is the supported path.
